src/puppetlabs/puppetdb/utils.clj

```diff
@@ -310,6 +311,12 @@
                 #(update % 0 underscores->dashes)
                 m))
 
+(defn synthesize-body-str
+  [{:strs [command version certname payload]}]
+  (json/generate-string
+    {:command (string/replace command "-" " ") :version (Integer. version)
```

Comment:
It looks like we're doing this twice, in `validate-command...` and here, is it possible to consolidate?